### DISCUSSION TITLE
how i calculated its-human reliability in EL1000

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#219] [Show and tell]
In #188 we learned that we should set up superdatasets to do analyses. In this S&T, I'm illustrating how I did that to estimate how AVC and CVC correlate across human and LENA (or VTC) annotations.

## Set up the analysis folder

```bash
source ~/ChildProjectVenv/bin/activate
datalad create -c yoda r_avc_cvc
cd r_avc_cvc
```

In order to share with others & get feedback, I followed [these instructions](https://childproject.readthedocs.io/en/latest/vandam.html#publish-to-gin) to publish to GIN, except that in the first step, I created the repo inside EL1000.

Initially, I made a mistake and created the repo in my individual profile (which would have meant I had to share it by hand with people in the EL1000 world), and also in EL1000, giving the name "gin" to the first one and "gin2" to the second one. I'm sure I'll find this confusing in the future -- I should have use "gin" for the EL1000 one. So to fix this, I had to:

```bash
datalad   siblings --name gin remove #remove the sibling name gin that I had created
datalad siblings add --name gin --url git@gin.g-node.org:/EL1000/r_avc_cvc.git # associate the name gin to the right ssh
datalad   siblings --name gin2 remove #remove the sibling name gin2 that I had created
```

Now that means that when I'm ready to share my progress, I do:

```bash
datalad save . -m "my commit message" # don't forget the dot!
datalad push --to gin  # don't forget the "to" bit
```

Once I set that up, however, my analyses got a permission error when I tried to write a file:
> PermissionError: [Errno 13] Permission denied: 'output/bergelson_cvc.csv'

To fix that I did: `datalad unlock output`
I could also have done: `datalad run -o output "python code/getr.py"`

## NOTE!

**Whenever there are updates** see the end of the document!!:

## Get the its data

```bash
datalad install -d . -r git@gin.g-node.org:/EL1000/EL1000.git  
cd EL1000
for j in */ ; do cd $j ; datalad run-procedure setup ; cd .. ; done
datalad get */annotations/its/converted . -J 4
```

## Get the human data 
Getting the human annotations involves doing different things for the different datasets, but this may change in the future:

For bergelson, lucid, winnipeg, and warlaumont, annotations are in a different branch:

```bash
cd bergelson
git fetch
git checkout main
datalad get annotations/eaf/an1/converted -J 4
cd ..
```

```bash
cd lucid
git fetch
git checkout eaf
datalad get annotations/eaf/an1/converted -J 4
cd ..
```

```bash
cd winnipeg
git fetch
git checkout eaf
datalad get annotations/eaf/an1/converted -J 4
cd ..
```

```bash
cd warlaumont
git fetch
git checkout eaf
datalad get annotations/eaf/annotator1/converted -J 4
cd ..
```

```bash
cd png2019
datalad get annotations/eaf/mc/converted -J 4
cd ..
```

```bash
cd vandam
datalad get annotations/cha/an1/converted -J 4
cd ..
```

```bash
cd lyon
datalad get annotations/cha/an1/converted -J 4
cd ..
```

```bash
cd tsimane2017
datalad get annotations/eaf/nk/converted -J 4
cd ..
```

## Run script

I looked at the following examples:
- [sample script](https://github.com/LAAC-LSCP/ChildProject/blob/master/examples/agreements.py)
- [docs](https://childproject.readthedocs.io/en/latest/api-metrics.html)
- [sample analysis](https://gin.g-node.org/EL1000/metrics)
- [sample script](https://github.com/LAAC-LSCP/ChildProject/discussions/211#discussioncomment-777326)

And with some help with Lucas, generated this python script that extracts the key sections of the annotations:

```
#!/usr/bin/env python3

from ChildProject.projects import ChildProject
from ChildProject.annotations import AnnotationManager
from ChildProject.metrics import gamma, segments_to_grid, grid_to_vector, vectors_to_annotation_task

import argparse

parser = argparse.ArgumentParser(description = 'compute correlation in number of vocs between its and human annotator')
parser.add_argument('--human', help = 'name of the set annotated by human')
parser.add_argument('--corpus', help = 'name of the corpus')
args = parser.parse_args()


project = ChildProject("EL1000/"+args.corpus) 
#project = ChildProject("EL1000/bergelson/")
#project = ChildProject("EL1000/png2019/")
am = AnnotationManager(project)
am.read()


intersection = AnnotationManager.intersection(am.annotations, ['its', args.human]) 
#intersection = AnnotationManager.intersection(am.annotations, ['its', "eaf/an1"])
#intersection = AnnotationManager.intersection(am.annotations, ['its', "eaf/mc"])
shared = intersection[intersection['set'] == args.human] 
shared.to_csv('output/' + args.corpus + '_shared.csv')

segments = am.get_collapsed_segments(intersection)



chi = segments[segments['speaker_type'] == 'CHI'] 

chi.to_csv('output/' + args.corpus + '_chi.csv')


cvc = chi.groupby(['position', 'set']).agg(
    voc = ('segment_onset', lambda x:len(x)),
    recording_filename = ('recording_filename', lambda x: x.iloc[0]),
    range_onset = ('range_onset', lambda x: x.iloc[0]),
).reset_index()


cvc = cvc.pivot(index = 'position', columns = 'set', values = ['voc','recording_filename','range_onset'])

cvc.fillna(0, inplace=True)
cvc.to_csv('output/' + args.corpus + '_cvc.csv')


adults = ['FEM', 'MAL']

ads = segments[segments['speaker_type'].isin(adults)] 
ads.to_csv('output/' + args.corpus + '_ads.csv')

avc = ads.groupby(['position', 'set']).agg(
    voc = ('segment_onset', lambda x:len(x)),
    recording_filename = ('recording_filename', lambda x: x.iloc[0]),
    range_onset = ('range_onset', lambda x: x.iloc[0]),
).reset_index()


avc = avc.pivot(index = 'position', columns = 'set', values = ['voc','recording_filename','range_onset'])

avc.fillna(0, inplace=True)
avc.to_csv('output/' + args.corpus + '_avc.csv')

```

I also wrote an R script, that pulls the actual statistics for each corpus:

```
#!/usr/bin/env Rscript


library("optparse")

option_list = list(
  make_option(c("-c", "--corpus"), type="character", default=NULL, 
              help="corpus name", metavar="character"),
  make_option(c("-h", "--human"), type="character", default="out.txt", 
              help="name of the set containing human annotations", metavar="character")
); 

opt_parser = OptionParser(option_list=option_list,add_help_option=FALSE);
opt = parse_args(opt_parser);


rep.row<-function(x,n){
  matrix(rep(x,each=n),nrow=n)
}



#human="eaf/an1"
#corpus="bergelson"
human=opt$human
corpus=opt$corpus


#get human annotations
read.csv(paste0("EL1000/", corpus, "/metadata/annotations.csv")) -> annos
subset(annos, set == human) -> human_annos
human_annos$unique = paste(human_annos$recording_filename, human_annos$range_onset)
n_human_anno=length(human_annos$unique)

#get shared annotation chunks
read.csv(paste0("output/", corpus, "_shared.csv"), header = T) -> shared
shared$unique = paste(shared$recording_filename, shared$range_onset)
n_shared=length(shared$unique)

sink(file=paste0("output/", corpus, "_output.txt"))
#report
print(
  "the following sections have been annotated by humans but not been matched with an its section"
)
unmatched = human_annos$unique[!(human_annos$unique %in% shared$unique)]
print(unmatched)

#cvc
read.csv(paste0("output/", corpus, "_cvc.csv"),
         header = F,
         skip = 3) -> cvc

colnames(cvc) <-
  c("position",
    "its",
    "human",
    "recording_filename",
    "x",
    "range_onset",
    "y")
cvc$unique = paste(cvc$recording_filename, cvc$range_onset)
print("**CVC ana**")

print("the following sections have zero segments (either not matched, or simply no segments)")
nosegs = human_annos$unique[!(human_annos$unique %in% cvc$unique)]
print(nosegs)
n_nosegs_cvc=length(nosegs)

print("correlation excluding clips where both its and human tagged zero segments")
x=cor.test(cvc[, c("its")], cvc[, c("human")])
print(x)
cvc_no0=x$cor

#add as zero those with nosegs but not in unmatched
toadd_names = nosegs[!(nosegs %in% unmatched)]

toadd = rep.row(cbind(NA, 0, 0, NA, NA, NA, NA, NA), length(toadd_names))
colnames(toadd) <- colnames(cvc)

cvc = rbind(cvc, toadd)
print("correlation including all")
x=cor.test(cvc[, c("its")], cvc[, c("human")])
print(x)
cvc_all=x$estimate


#avc
read.csv(paste0("output/", corpus, "_avc.csv"),
         header = F,
         skip = 3) -> avc
colnames(avc) <-
  c("position",
    "its",
    "human",
    "recording_filename",
    "x",
    "range_onset",
    "y")
avc$unique = paste(avc$recording_filename, avc$range_onset)
print("**AVC ana**")
print("the following sections have zero segments (either not matched, or simply no segments)")
nosegs = human_annos$unique[!(human_annos$unique %in% avc$unique)]
(print(nosegs))
n_nosegs_avc=length(nosegs)


print("correlation excluding clips where both its and human tagged zero segments")
x=cor.test(avc[, c("its")], avc[, c("human")])
print(x)
avc_no0=x$estimate

toadd = rep.row(cbind(NA, 0, 0, NA, NA, NA, NA, NA), length(toadd_names))
colnames(toadd) <- colnames(avc)

avc = rbind(avc, toadd)
print("correlation including all")
x=cor.test(avc[, c("its")], avc[, c("human")])
print(x)
avc_all=x$estimate

sink()

print(paste(corpus, human, n_human_anno, n_shared, length(unmatched), n_nosegs_cvc, n_nosegs_avc, cvc_no0, cvc_all, avc_no0, avc_all))

```

I then ran both scripts from the top level using these commands:

```
python3 code/getr.py  --human eaf/an1 --corpus bergelson
./code/getr2.R --human eaf/an1 --corpus bergelson
python3 code/getr.py --human eaf/an1 --corpus lucid
./code/getr2.R --human eaf/an1 --corpus lucid
python3 code/getr.py --human eaf/an1 --corpus winnipeg
./code/getr2.R --human eaf/an1 --corpus winnipeg
python3 code/getr.py  --human eaf/mc --corpus png2019
./code/getr2.R --human eaf/mc --corpus png2019

```

**missing**
python3 code/getr.py --human cha/an1 --corpus vandam
./code/getr2.R --human cha/an1 --corpus vandam
python3 code/getr.py  --human eaf/annotator1 --corpus warlaumont

**corpus quarantined**
python code/getr.py  --human cha/an1 --corpus lyon

### Error after update

Sometimes I get
> Traceback (most recent call last):
  File "code/getr.py", line 25, in <module>
    shared = intersection[intersection['set'] == args.human] 
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/frame.py", line 2906, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2897, in get_loc
    raise KeyError(key) from err
KeyError: 'set'

when running getr.py. When that happens, it get fixed by:

- navigating to the relevant folder
- git pull -- if it says that no branch has been defined:
- check which is the relevant branch on gin, then git checkout there
- git pull
- datalad get annotations/... etc

### Picking up where I left off

Some of the datasets needed to be fixed in some way. To pick up again from the same spot but updating the databases I did:

```bash
datalad update merge . -r # to update my own code as well as all the nested datasets
# navigate to the analysis folder, then
source ~/ChildProjectVenv/bin/activate # activate the environment if needed
cd EL1000/png2019 # change to the dataset you want to update
datalad update --merge
datalad get annotations/its/converted/
datalad get annotations/eaf/mc/converted/
```
Once in a while, you can instead (or in addition) update the package

```bash
pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git #to update the package
```

Note that by doing `datalad update --merge` I'm taking into account all the siblings. If I didn't care about getting the most recent best version, I could have simply done `git pull` which gets an update to the precise branch I'm in.

Even so, you may need to sometimes do `datalad get annotations/its/converted` and the counterpart for the human annotations.

git submodule update --init --recursive --remote

### vandam OK
But check with Lucas about cha import

### lucid, winnipeg, lyon 
> KeyError: 'set'
(ChildProjectVenv) (base) Mac-mini:r_avc_cvc acristia$ python3 code/getr.py  --human eaf/annotator1 --corpus warlaumont
(ChildProjectVenv) (base) Mac-mini:r_avc_cvc acristia$ python3 code/getr.py  --human eaf/mc --corpus png2019
(ChildProjectVenv) (base) Mac-mini:r_avc_cvc acristia$ python code/getr.py  --human cha/an1 --corpus lyon
Traceback (most recent call last):
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2895, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1675, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1683, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'set'
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "code/getr.py", line 25, in <module>
    shared = intersection[intersection['set'] == args.human] 
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/frame.py", line 2906, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2897, in get_loc
    raise KeyError(key) from err
KeyError: 'set'

**Comment by alecristia** [comment]
## how I replicated the analyses in a new computer

First I made sure that I had datalad and childproject installed, following these instructions:
https://childproject.readthedocs.io/en/latest/install.html

Then I got a copy of the repo & contents

source ~/ChildProjectVenv/bin/activate
datalad install -r git@gin.g-node.org:/EL1000/r_avc_cvc.git

**Reply by alecristia** [reply]
mac-cristia:gitrepos alejandrinacristia$ source ~/ChildProjectVenv/bin/activate
(ChildProjectVenv) mac-cristia:gitrepos alejandrinacristia$ datalad install -r git@gin.g-node.org:/EL1000/r_avc_cvc.git
Clone attempt:   0%|              | 0.00/1.00 [00:00<?, ? Candidate locations/s]The authenticity of host 'gin.g-node.org (141.84.41.219)' can't be established.
ECDSA key fingerprint is SHA256:E35RRG3bhoAm/WD+0dqKpFnxJ9+yi0uUiFLi+H/lkdU.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'gin.g-node.org,141.84.41.219' (ECDSA) to the list of known hosts.
git@gin.g-node.org: Permission denied (publickey,keyboard-interactive).
[ERROR  ] Failed to clone from all attempted sources: ['git@gin.g-node.org:/EL1000/r_avc_cvc.git'] [install(/Users/alejandrinacristia/gitrepos/r_avc_cvc)] 
install(error): /Users/alejandrinacristia/gitrepos/r_avc_cvc (dataset) [Failed to clone from all attempted sources: ['git@gin.g-node.org:/EL1000/r_avc_cvc.git']]
[ERROR  ] cmd:git of version >= 2.19.1 is missing. You have version 2.14.1 [external_versions.py:check:375] (OutdatedExternalDependency)

shows two problems:

no access, which means I forgot to add my ssh key into GIN, as specified in [here](https://gin.g-node.org/EL1000/el1000#configuring-your-ssh-key-on-gin)

and also, git too young:

` brew upgrade git`

**Reply by alecristia** [reply]
ERROR: Enter passphrase for key '/Users/alejandrinacristia/.ssh/id_rsa': 

I couldn't figure this one out -- I guess I should perhaps generate a new one (but why is it asking for a pass phrase, and why does it refer to id_rsa rather than pub?) but that would entail also redoing my connection with oberon and flores

**Comment by alecristia** [comment]
remember, if you get errors like:
FileNotFoundError: [Errno 2] No such file or directory: 'EL1000/lyon/annotations/cha/an1/converted/COF_030503_1_0_0.csv'

the most likely reason is that you forgot to get annotations!

**Comment by alecristia** [comment]
If you can already ssh into Oberon with an alias by doing `ssh foberon` or `ssh oberon` or similar, skip this step. Otherwise, follow these instructions to setup ssh properly

1. Add the following lines to your `~/.ssh/config` file, replacing `<username>` by your actual username:
ost flores
   Hostname cognitive-ml.fr
   User acristia

Host oberon
   Hostname oberon
   User acristia
   ProxyJump flores


Host *
 AddKeysToAgent yes
 UseKeychain yes
 IdentityFile ~/.ssh/id_rsa
 ForwardX11 yes


2. Try to SSH into oberon by typing `ssh foberon`. You will be prompted for your credentials.

**Comment by alecristia** [comment]
i had to move to a different computer, so I did:

- activate my env `source ~/ChildProjectVenv/bin/activate`
- check that git-annex, datalad, and package are installed
- check that I have the latest version of the childproject package `pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git`
- install the package with everything within `datalad install -r git@gin.g-node.org:/EL1000/r_avc_cvc.git`
- i had a permissions problem:
> git@gin.g-node.org: Permission denied (publickey,keyboard-interactive).
[ERROR  ] Failed to clone from all attempted sources: ['git@gin.g-node.org:/EL1000/r_avc_cvc.git'] [install(/Users/acristia/Documents/gitrepos/r_avc_cvc)] 
install(error): /Users/acristia/Documents/gitrepos/r_avc_cvc (dataset) [Failed to clone from all attempted sources: ['git@gin.g-node.org:/EL1000/r_avc_cvc.git']]
[ERROR  ] No installed dataset found at /Users/acristia/Documents/gitrepos/r_avc_cvc. [dataset.py:require_dataset:583] (ValueError) 

- this gets fixed by checking that my ssh has been declared on gin: https://gin.g-node.org/user/settings/ssh
- set up the data folders & get the converted its:

```
cd r_avc_cvc/EL1000
for j in */ ; do cd $j ; datalad run-procedure setup ; cd .. ; done
datalad get */annotations/its/converted . -J 4
```
- download the human data following the instructions above (not repeated here since they make reference to branches, which may change in the future)

**Comment by alecristia** [comment]
There's a new set of annotations for png.

I installed that dataset in oberon, then I got the its annotations.

The new eaf annotations had not been added or imported. I put them in the raw folder, removed the converted versions (bc they do not correspond to the raw ones), and pushed git commit message "new annotations".

I then went to import by running `python scripts/import_eaf.py` (after having changed the tier that needs to be used from recording to final). This gave me an error because we need the raw its for this import.
When I tried to get, I got "[ERROR  ] not available" -- although these are clearly there (eg [here](https://gin.g-node.org/LAAC-LSCP/png2019/src/main/annotations/its/confidential/raw/e20190820_092711_013099.its)).

So probably the install of this dataset was shoddily done? It doesn't look like it: 
`$ datalad siblings`

>.: here(+) [git]
.: origin(+) [git@gin.g-node.org:/EL1000/png2019.git (git)]

I tried getting a non-confidential file, and that worked fine.

So this suggests a permissions issue -- since it's confidential files I'm trying to get...

**Reply by lucasgautheron** [reply]
Hello,

In order to maintain LAAC datasets shared in EL1000; currently the right way to do that is to install them from their LAAC url, run the setup procedure and then push your work to any of the relevant siblings;

```bash
# setup
datalad install git@gin.g-node.org:/LAAC-LSCP/png2019.git
cd png2019
datalad run-procedure setup --private --confidential

# do work
datalad save annotations/newset/raw

# publish to LAAC-LSCP/png2019
datalad push --to origin

# share these annotations with EL1000
git annex metadata --set EL1000=yes annotations/newset/raw --force
datalad push --to EL1000
```

**Reply by alecristia** [reply]
ah right! so the problem was that my siblings did not include a confidential one... so I ran

`datalad run-procedure setup --private --confidential`

and now it IS properly listed:

`$ datalad siblings`

>.: here(+) [git]
.: origin(+) [git@gin.g-node.org:/EL1000/png2019.git (git)]
.: confidential(+) [git@gin.g-node.org:/EL1000/png2019-confidential.git (git)]
.: private(+) [git@gin.g-node.org:/LAAC-LSCP/png2019.git (git)]

and now I can get the raw files.

thank you!

**Reply by alecristia** [reply]
ah i should be careful -- I need to change "origin" here, bc it points to EL1000

to this end, I need to:

```
datalad siblings configure -s origin --url git@gin.g-node.org:/LAAC-LSCP/png2019.git
datalad siblings add -s EL1000 --url git@gin.g-node.org:/EL1000/png2019.git 
datalad siblings
```

> .: here(+) [git]
.: confidential(+) [git@gin.g-node.org:/EL1000/png2019-confidential.git (git)]
.: private(+) [git@gin.g-node.org:/LAAC-LSCP/png2019.git (git)]
.: EL1000(+) [git@gin.g-node.org:/EL1000/png2019.git (git)]
.: origin(+) [git@gin.g-node.org:/LAAC-LSCP/png2019.git (git)]

That's still not right... confidential is pointing to EL1000, which it shouldn't, right?

So I tried to run the procedure again:

```
datalad run-procedure setup --private --confidential
datalad siblings
```
> .: here(+) [git]
.: el1000(+) [git@gin.g-node.org:/EL1000/png2019.git (git)]
.: confidential(+) [git@gin.g-node.org:/EL1000/png2019-confidential.git (git)]
.: origin(+) [git@gin.g-node.org:/LAAC-LSCP/png2019.git (git)]
.: private(+) [git@gin.g-node.org:/LAAC-LSCP/png2019.git (git)]
.: EL1000(+) [git@gin.g-node.org:/EL1000/png2019.git (git)]

the 2 EL1000 are confusing, but based on your code, it looks like it's the allcaps I should keep, not the lowercaps -- but perhaps it makes no difference...

what a mess I'm making! I think you're right, I should have uninstalled and re-installed from the right path...

**Reply by alecristia** [reply]
I'm trying to replicate analyses in oberon, and I think everything is now installed correctly, but I'm getting this error after a while of computing:

> See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = _infer_fill_value(value)
Traceback (most recent call last):
  File "code/reliability.py", line 128, in <module>
    compute_reliability(sample) for sample in samples.to_dict(orient = 'records')
  File "code/reliability.py", line 128, in <listcomp>
    compute_reliability(sample) for sample in samples.to_dict(orient = 'records')
  File "code/reliability.py", line 65, in compute_reliability
    cvc = cvc.pivot(index = ['recording_filename', 'range_onset'], columns = 'set', values = ['voc'])
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/frame.py", line 6678, in pivot
    return pivot(self, index=index, columns=columns, values=values)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/pivot.py", line 477, in pivot
    return indexed.unstack(columns)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/frame.py", line 7146, in unstack
    return unstack(self, level, fill_value)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 418, in unstack
    return _unstack_frame(obj, level, fill_value=fill_value)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 439, in _unstack_frame
    obj.index, level=level, constructor=obj._constructor,
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 120, in __init__
    self._make_selectors()
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 169, in _make_selectors
    raise ValueError("Index contains duplicate entries, cannot reshape")
ValueError: Index contains duplicate entries, cannot reshape

By the way, `datalad status`

> untracked: output/lyon_ads.csv (file)
untracked: output/lyon_chi.csv (file)
untracked: output/lyon_shared.csv (file)

Just in case, I'm launching the same analysis but now giving a path that doesn't exist: `python code/reliability.py --output output2/ --download` (output2 instead of output)

**Reply by alecristia** [reply]
no, it wasn't that -- same error

> Traceback (most recent call last):
  File "code/reliability.py", line 128, in <module>
    compute_reliability(sample) for sample in samples.to_dict(orient = 'records')
  File "code/reliability.py", line 128, in <listcomp>
    compute_reliability(sample) for sample in samples.to_dict(orient = 'records')
  File "code/reliability.py", line 65, in compute_reliability
    cvc = cvc.pivot(index = ['recording_filename', 'range_onset'], columns = 'set', values = ['voc'])
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/frame.py", line 6678, in pivot
    return pivot(self, index=index, columns=columns, values=values)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/pivot.py", line 477, in pivot
    return indexed.unstack(columns)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/frame.py", line 7146, in unstack
    return unstack(self, level, fill_value)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 418, in unstack
    return _unstack_frame(obj, level, fill_value=fill_value)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 439, in _unstack_frame
    obj.index, level=level, constructor=obj._constructor,
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 120, in __init__
    self._make_selectors()
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/reshape.py", line 169, in _make_selectors
    raise ValueError("Index contains duplicate entries, cannot reshape")
ValueError: Index contains duplicate entries, cannot reshape

**Reply by lucasgautheron** [reply]
I'll try to reproduce ; it is possible that some breaking change to one of the datasets may have happened...

**Comment by alecristia** [comment]
also, it looks like we may revisit the code -- if so, it would be great to store a matrix that has AVC and CVC according to system and human for each clip? that will help us find errors of eg alignment (if the numbers are very close except for a few outliers)

**Reply by lucasgautheron** [reply]
Will do that!
However, for spotting errors, I think the best way is to use the graphical inspector - and it is very fast too.

**Comment by lucasgautheron** [comment]
I've improved the documentation, updated the version of EL1k used as a dependency, enabled multiprocessing for faster processing.

The following instructions should work. Can you please tell me if they still do not work for you?

1.
```bash
$ datalad install git@gin.g-node.org:/EL1000/reliability.git -r
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability (dataset)                                                                                                                                               
[INFO   ] Installing Dataset(/scratch2/lgautheron/tests/reliability) to get /scratch2/lgautheron/tests/reliability recursively 
[INFO   ] scanning for unlocked files (this may take some time) 
install(ok): /scratch2/lgautheron/tests/reliability/EL1000 (dataset)                                                                                                                                        
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/alphen (dataset)                                                                                                                                 
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/bergelson (dataset)                                                                                                                              
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/elo (dataset)                                                                                                                                    
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/ganek (dataset)                                                                                                                                  
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/kalashnikova (dataset)                                                                                                                           
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/kidd (dataset)                                                                                                                                   
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/lucid (dataset)                                                                                                                                  
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
install(ok): /scratch2/lgautheron/tests/reliability/EL1000/lyon (dataset)                                                                                                                                   
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
[INFO   ] scanning for unlocked files (this may take some time)                                                                                                                                             
  [8 similar messages have been suppressed]                                                                                                                                                                 
action summary:                                                                                                                                                                                             
  install (ok: 18)
```
2.
```bash
(base) [lgautheron@oberon reliability]$ sbatch -c 32 -o testrun code/reliability.py --output output/testrun.csv --threads 32 --download
Submitted batch job 175603
```

3.
```bash
datalad get output/reliability.csv
diff output/testrun.csv output/reliability.csv
```

**Reply by alecristia** [reply]
double checking: should I update png that is inside input/ here, to make sure the most recent versions of the eaf files are considered? The r for CVC doesn't seem to have changed...

**Comment by lucasgautheron** [comment]
The values have changed, but I pushed the latest results already. So it should be good
